### PR TITLE
Added tags with search function to diary entries

### PR DESCRIPTION
**Pull request by BrandtM**
Hey.  
I noticed that tags were on your ToDo list and I thought I'd give it a shot.  
It's nothing groundbreaking of course but it works and I am quite pleased with the result and the project as a whole.  
I tried to stick to your design choices as close as possible.  

Basically it is now possible to specify whether you want to search for content or tags and during diary creation it is possible to add tags.  
I should note that I specify tags as "comma separated tags" in the script. In theory I don't think this has to be the case but it should be done like so anyway to maintain readability.  
Tags are also optional but this I did not include in the script. I made sure that the script doesn't break if tags were not specified but it is still up to the user to include them.

### CODE REVIEW

**Comment by prodicus** [comment]
Hey @BrandtM .

Thanks for contributing! Yes the tags were one of the first feature requests made (#1 )

I noticed two problems here. When searching for something in

- **tags**
We get an error if the search query is not present there in the **db**

It throws an error something like this

```sh
Traceback (most recent call last):
  File "./journal.py", line 164, in <module>
    menu_loop()
  File "./journal.py", line 55, in menu_loop
    menu[choice]()
  File "./journal.py", line 135, in search_entries
    view_entry(input("Enter a search Query: "), search_content=False)
  File "./journal.py", line 88, in view_entry
    entry = entries[index]
IndexError: list index out of range
```

- **Content**

Same case when we try to search for something we don't have in the **db**.

```sh
Traceback (most recent call last):
  File "./journal.py", line 164, in <module>
    menu_loop()
  File "./journal.py", line 55, in menu_loop
    menu[choice]()
  File "./journal.py", line 138, in search_entries
    view_entry(input("Enter a search Query: "), search_content=True)
  File "./journal.py", line 88, in view_entry
    entry = entries[index]
IndexError: list index out of range
```

On a side note. We could make the search look something like this. 

```sh
What do you want to search for?
c) Content
t) Tags
q) Return to the main menu
=======================
Action [c/t/q] : 
```
Looks much more cleaner and is in consistency with the other methods. What do you say?

**Comment by BrandtM** [comment]
Hello.  
Thanks for your feedback. I will definitely add some safety later.  

Also I think that making the menu look like this:  
```
What do you want to search for?
c) Content
t) Tags
q) Return to the main menu
=======================
Action [c/t/q] : 
```  

is definitely the way to go.  

I'll keep you posted.

**Comment by prodicus** [comment]
No problem at all! Anticipating a PR from you soon then :+1:

**Comment by BrandtM** [comment]
Made some changes. Looks way better now haha.  
The search was also improved so that it doesn't throw a fit if there are no results.

**Comment by prodicus** [comment]
Awesome!

Will merge after testing it on my Linux box @BrandtM . What system are you testing it on by the way?

**Comment by BrandtM** [comment]
@prodicus I'm developed and tested this on Ubuntu 15.04 right now.  
But later I will switch to Windows 10.

**Comment by prodicus** [comment]
:+1:

File: journal.py [hunk 1]
```diff
@@ -118,7 +130,26 @@ def view_entry(search_query=None):
...
     """Let's us search through the diary entries"""
-    view_entry(input("Enter a search Query: "))
+    while 1:
+        clear()
+        print("What do you want to search for?")
+        print("c) Content")
+        print("t) Tags")
+        print("q) Return to the main menu")
+        print("===============================")
+        print("Action [c/t/q]: ", end="")
```
Comment (prodicus) [inline journal.py]: I was adding **python2** support and this line throws me a 

    SyntaxError: invalid syntax

Could you elaborate on what exactly does it do?

**Comment by BrandtM** [comment]
Ah I think the error is in this line  
```python
print("Action [c/t/q]: ", end="")
```  

What this does is instead of automatically terminating the print() output with a newline, terminate it with nothing. This means that the user can actually type after the "Action.." line instead of being on a new line.  

I found [this Stack Overflow answer](http://stackoverflow.com/questions/493386/how-to-print-in-python-without-newline-or-space/493399#493399) which provides a method to achieve this in Python 2.x

**Comment by prodicus** [comment]
:+1: On it

**Pull request merged**